aiidalab_widgets_base/process.py

```diff
@@ -102,13 +109,19 @@ def follow(self, detach=False):
             update_state.start()
         else:
             self._follow()
+            for func in self._run_after_completed:
+                func(self.process)
+
+    def run_after_completed(self, function):
+        """Run functions after a process has been completed."""
+        if callable(function):
+            self._run_after_completed.append(function)
```

Comment:
I don't think it necessary to check the argument type here. If the user provides something that is not callable, they will notice.

Reply:
agreed.